tests/lang/s02_core/s02_procedures/t03_overload_core_subtype_bug.nim

```diff
@@ -0,0 +1,42 @@
+discard """
+description: '''
+Rejects valid code, or this part of the overload resolution is underspecified.
+
+`ref object` in definition is handled differently from `ref T` in the argument
+when overload resolution is considered.
+'''
+knownIssue: "ref object is not a proper subtype"
+errmsg: "Error: type mismatch: got <ref Derived>"
+
+"""
+
+# When fixed should be added back to the core overload resolution, "subtype" section
+block subtype_ref_object:
+  block no_qualifier:
+    ## Identical to previous example, but used here to highlight lack of
+    ## difference in handling of these cases.
+    type
+      Base = object of RootObj
+      Derived = object of Base
+
+    proc impl(arg: Base) = discard
+
+    impl(Derived())
+
+  block as_qualifier:
+    type
+      Base = object of RootObj
+      Derived = object of Base
+
+    proc impl(arg: ref Base) = discard
+
+    impl((ref Derived)(nil))
+
+  block in_definition:
```

Comment:
Related, although not fully identical issue - https://github.com/nim-lang/Nim/issues/11136 it was closed because it "works as expected", but I'm not sure if this is a correct way to handle this. 

This compiles, works (does not crash) and produces expected result? (`(field: 12, afield: 1222)`). Embedded part of the supertype is passed, modified and functions properly

```nim
type
  A = ref object of RootObj
    afield: int
    
  B = ref object of A
    field: int

proc init(a: var A) =
  a = A(afield: 1222)

var b = B(field: 123)
b.A.init() 
echo b[]
```

Reply:
It was decided that this particular example conforms to the spec because `var ref Derived` is not a subtype of `Base`. `var Derived` is, `ref Derived` and `ptr Derived` are subtypes as well, but double indirection makes it a non-subtype.

Reply:
It was decided that this particular example conforms to the spec because `var ref Derived` is not a subtype of `Base`. `var Derived` is, `ref Derived` and `ptr Derived` are subtypes as well, but double indirection makes it a non-subtype.

Reply:
`ref object` being different from `ref` of the `object` is probably a bug though, otherwise there is a semantic difference between `type Obj = ref object` + `arg: Obj` and `type Obj = object` + `arg: ref Obj`, which does not makes sense to me, at least not now.

Reply:
That last one definitely doesn't make sense to me.

I think of it as doing the `ref` op on a type should return a ref type, don't really know how else it could work? Maybe it's to do with the silly auto dereference feature? 🤔